db.go

```diff
@@ -249,3 +263,12 @@ func readEntryLengths(reader *bufio.Reader) (int, int) {
 	valLen, _ := binary.ReadUvarint(reader)
 	return int(keyLen), int(valLen)
 }
+
+func (db *DB) OpenSSTableByFileName(fileName string) (*SSTable, error) {
+	file, err := os.Open(fileName)
```

Comment:
### Runtime Errors

- **File Opening Error Handling**: In `OpenSSTableByFileName`, ensure that the file is properly closed if an error occurs during the opening process. This can be achieved using a `defer` statement:

```go
file, err := os.Open(fileName)
if err != nil { 
    return nil, err 
}
defer file.Close()
return db.OpenSSTable(file)
```